components/script/dom/resizeobserver.rs

```diff
@@ -96,8 +97,10 @@ impl ResizeObserver {
         can_gc: CanGc,
     ) {
         let mut entries: Vec<DomRoot<ResizeObserverEntry>> = Default::default();
-        for (observation, target) in self.observation_targets.borrow_mut().iter_mut() {
-            let ObservationState::Active(box_size) = observation.state else {
+        let observation_targets = self.observation_targets.borrow();
+        for (observation, target) in observation_targets.iter() {
+            let mut observation_state = observation.state.borrow_mut();
```

Comment:
This mutable borrow has the same problem as the original issue—it's still active when we reach `let observer_size = ResizeObserverSize::new(&window, size_impl, can_gc);`.

Reply:
updated 👍